algebra/Cargo.toml

```diff
@@ -47,3 +47,10 @@ mnt6_753 = []
 std = [ "algebra-core/std" ]
 parallel = [ "std", "algebra-core/parallel" ]
 derive = [ "algebra-core/derive" ]
+asm = [ "algebra-core/asm" ]
+
+full_asm = [ "algebra-core/asm", "bls12_377", "bls12_381", "sw6", "mnt4_298", "mnt4_753", "mnt6_298", "mnt6_753", "edwards_bls12", "edwards_sw6", "jubjub" ]
+small_asm = ["algebra-core/asm", "mnt4_298", "mnt6_298" ]
+mid_asm = [ "algebra-core/asm", "bls12_377", "bls12_381", "edwards_bls12"]
+big_asm = [ "algebra-core/asm", "sw6", "mnt4_753", "mnt6_753", "edwards_sw6" ]
+mix_asm = [ "algebra-core/asm", "sw6", "mnt4_753", "bls12_381", "mnt6_298" ]
```

Comment:
I don't think we need these features; because features are additive, if you enable `asm` and any of these other features, the underlying arithmetic will automatically use the asm routines.

Reply:
I don't think that is the case. At least not with the current `cfg` logic. If I try to run 
```
RUSTFLAGS="--emit=asm -C target-cpu=native -C target-feature=+bmi2,+adx" cargo +nightly test --features asm bls12_381
```
none of the tests run.